shields: stm32: avoid LTDC format and HEAP size settings in shields conf

avolmat-st opened the pull request:
The LTDC pixel format and HEAP size is a matter of the application rather than the configuration of the shield. Remove such config from the stm32 related (LTDC) display shields and as an example, move them into the display sample app.

Commit:
shields: stm32: avoid LTDC format and HEAP size settings in shields conf

The LTDC pixel format and HEAP size is a matter of the application
rather than the configuration of the shield.
The LTDC has already a default RGB565 configuration hence such conf
is not more necessary, moreover sample display heap rework make the
CONFIG_HEAP_MEM_POOL_SIZE setting no more necessary.

cf "samples: display: define HEAP_MEM_POOL_ADD_SIZE_ for display buffer"

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

facchinm reviewed (dismissed):
It looks like the patch is changing `CONFIG_STM32_LTDC_ARGB8888` to `CONFIG_STM32_LTDC_RGB565` for both the boards and not considering to create the same for `arduino_giga_r1` target. Please make sure that this is the intended behaviour since the framebuffer occupation halves. And consider moving the config for the Giga too (maintaining CONFIG_STM32_LTDC_RGB565=y)

avolmat-st commented:
Converted back to draft to avoid polluting. I will update missing bits and push it again later.

Commit:
shields: st_b_lcd40_dsi1_mb1166: use RGB565 as default format

The RGB565 format should be suffisant as default format of LTDC
when using this shield and allows to keep the default framebuffer
memory usage low.
Application requiring RGB888 or ARGB8888 input format can later
on update the LTDC input format when really necessary.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
st: lvgl: move all STM32 common LVGL settings in soc Kconfig.defconfig

LVGL BITS_PER_PIXEL and COLOR_DEPTH are closely linked to how the LTDC
is configured on the platform. Move all all STM32 common LVGL
configuration into the common stm32 Kconfig.defconfig.

CONFIG_LV_DPI_DEF and CONFIG_INPUT are application related.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

erwango commented on samples/drivers/display/boards/stm32h747i_disco_stm32h747xx_m7.conf:
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2025 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_HEAP_MEM_POOL_SIZE=65535
```
Should it be cleaned up from `boards/shields/st_b_lcd40_dsi1_mb1166/boards/stm32h747i_disco_stm32h747xx_m7.conf` then ?

Btw, you might also want to have a look to `boards/shields/st_lcd_dsi_mb1835/boards/stm32u5g9j_dk1.conf`

erwango reviewed (commented):
Otherwise LGTM

Commit:
shields: st_lcd_dsi_mb1835: use RGB565 as default format

The RGB565 format should be suffisant as default format of LTDC
when using this shield and allows to keep the default framebuffer
memory usage low.
Application requiring RGB888 or ARGB8888 input format can later
on update the LTDC input format when really necessary.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

avolmat-st commented on samples/drivers/display/boards/stm32h747i_disco_stm32h747xx_m7.conf:
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2025 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_HEAP_MEM_POOL_SIZE=65535
```
> Random though: What about using `HEAP_MEM_POOL_ADD_SIZE_` ? See https://docs.zephyrproject.org/latest/kernel/memory_management/heap.html#defining-the-heap-memory-pool

I think we will still need to define somehow a size of HEAP which is based on what the application does and the size of the display, so still it would I think require to have a dedicate conf file for each setup used with this sample.

jfischer-no reviewed (dismissed):
@erwango on what basis did you dismissed my review?

erwango commented:
> @erwango on what basis did you dismissed my review?

I dismissed _both_ reviews for which comments were addressed. And I mentioned you were free to have a new look.

Can you explain why you're blocking this PR now ?

> I do not think that this kind of single line overlays should be in Zephyr tree. However, the problem is more extensive, I opened https://github.com/zephyrproject-rtos/zephyr/issues/102912.

As you mention yourself, problem is more extensive and there are tons of similar examples today in zephyr and no guideline whatsoever to restrict it. 
You raised your concern in #102912, this is fine and will be discussed. But it may not be approved soon or not be approved at all because we won't agree on an alternate solution.
So, I think that the fact that you're blocking this PR is abusive. As assignee of this PR, I ask you to re-consider.

kartben commented:
Adding a DNM for the next 24 hours as per 
 https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#:~:text=To%20give%20the%20reviewers%20time%20to%20respond%20and%20escalate%2C%20the%20assignee%20is%20expected%20to%20block%20the%20PR%20from%20being%20merged%20either%20by%20not%20approving%20the%20PR%20or%20by%20setting%20the%20DNM%20label.

jfischer-no reviewed (changes requested):
> @jfischer-no I'm dismissing your review again, as I don't think blocking is justified. You're free to escalate.

Blocking is justified. It is you, who are free to escalate.

avolmat-st commented:
PR https://github.com/zephyrproject-rtos/zephyr/pull/103706 has now been merged hence rebased on top of it.
Is the Architecture Review label still necessary ?

Commit:
modules: lvgl: fix LV_Z_BITS_PER_PIXEL default value

Fix the order of default in the Kconfig LV_Z_BITS_PER_PIXEL
in order to be able to have the value set automatically based
on the COLOR_DEPTH.
Without that since the default 32 was coming first, 32 would
always be selected.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

avolmat-st commented on soc/st/stm32/Kconfig.defconfig:
```diff
@@ -126,4 +126,24 @@ config VIDEO_BUFFER_POOL_ALIGN
+config LV_Z_BITS_PER_PIXEL
+	default 16 if STM32_LTDC_RGB565
+	default 24 if STM32_LTDC_RGB888
+	default 32 if STM32_LTDC_ARGB8888
```
Yes, however I first had to fix the default value handling of LV_Z_BITS_PER_PIXEL and for that added a commit in the PR.

Commit:
boards: remove useless LV_Z_BITS_PER_PIXEL default setting

LVGL Kconfig already set the LV_Z_BITS_PER_PIXEL config based
on the LV_COLOR_DEPTH so avoid setting LV_Z_BITS_PER_PIXEL
for boards / shields when the LV_COLOR_DEPTH is also set.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

avolmat-st commented:
Since now LV_Z_BITS_PER_PIXEL default is properly computed based on COLOR_DEPTH, 1 more commit added to remove treewide the LV_Z_BITS_PER_PIXEL duplicated with LV_COLOR_DEPTH in board / shield Kconfigs.
